pkg/lazy/lazy.go

```diff
@@ -64,19 +71,24 @@ func (l *lazyImpl[T]) Get() (T, error) {
 func (l *lazyImpl[T]) doSlow() (T, error) {
 	l.m.Lock()
 	defer l.m.Unlock()
-	if l.done == 0 {
-		done := uint32(1)
+	if !l.done.Load() {
+		done := true
 		// Defer in case of panic
 		defer func() {
-			atomic.StoreUint32(&l.done, done)
+			l.done.Store(done)
 		}()
 		res, err := l.getter()
```

Comment:
I am not sure I agree with this. 

Sure, we might return non-nil T and non-nil error. But we are returning the result of `getter`. If the user *wants* to return that for whatever reason, they ought to be able to.

Its the callers responsibility to determine the semantics of their function, not a generic abstraction over it?

Reply:
I think it is similar to what we do on backoff wrap, it is to simplify the caller. 

In most cases, we users could think when err happens, the result would be empty and not go on use the result.